zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -318,6 +318,253 @@ def test_security_checklist(self):
+        if self.current_os_release < trusty_kilo:
+            logging.debug('Skipping test, sriov agent not supported on < '
+                          'trusty/kilo')
+            return
+
+        zaza.model.set_application_config(
+            self.application_name,
+            {'sriov-device-mappings': 'physnet42:eth42',
+             'enable-sriov': 'True'})
+
+        zaza.model.wait_for_application_states()
+
+        # Config file affected by juju set config change
+        conf_file = '/etc/neutron/plugins/ml2/sriov_agent.ini'
+        expected = {
+            'sriov_nic': {
+                'physical_device_mappings': ['physnet42:eth42'],
+            },
+        }
+
+        zaza.model.block_until_oslo_config_entries_match(
+            self.application_name,
+            conf_file,
+            expected
+        )
+
+        # the CI environment does not expose an actual SR-IOV NIC to the
+        # functional tests. consequently the neutron-sriov agent will not
+        # run, and the charm will update its status as such. this will prevent
+        # the success of pause/resume test.
+        #
+        # disable sriov after validation of config file is complete.
+        logging.info('Disabling SR-IOV after verifying config file data...')
+
+        zaza.model.set_application_config(
+            self.application_name,
+            {'sriov-device-mappings': '',
+             'enable-sriov': 'False'})
+
+        logging.info('Waiting for config-changes to complete...')
+        zaza.model.wait_for_application_states()
+
+        logging.debug('OK')
+
+    def _check_ml2_setting_propagation(self, service, charm_key,
+                                       config_file_key, vpair,
+                                       section):
+
+        set_default = {charm_key: vpair[0]}
+        set_alternate = {charm_key: vpair[1]}
+        app_name = service
+
+        conf_file = '/etc/neutron/plugins/ml2/openvswitch_agent.ini'
+
+        expected = {
+            section: {
+                config_file_key: [vpair[1]],
+            },
+        }
+
+        with self.config_change(set_default,
+                                set_alternate,
+                                application_name=app_name):
+            zaza.model.block_until_oslo_config_entries_match(
+                self.application_name,
+                conf_file,
+                expected,
+            )
+        logging.debug('OK')
+
+    def test_l2pop_propagation(self):
+        """Verify that l2pop setting propagates to neutron-ovs."""
+        # Needs love - not idempotent
+        self._check_ml2_setting_propagation('neutron-api',
+                                            'l2-population',
+                                            'l2_population',
+                                            ['False', 'True'],
+                                            'agent')
+
+    def test_nettype_propagation(self):
```

Comment:
This should be numbered. I know it is not in the original but it one of the tests.